src/BuiltInTools/dotnet-watch/Program.cs

```diff
@@ -147,6 +147,11 @@ internal static RootCommand CreateRootCommand(Func<CommandLineOptions, Task<int>
             root.TreatUnmatchedTokensAsErrors = false;
             root.Handler = CommandHandler.Create((CommandLineOptions options, ParseResult parseResults) =>
             {
+                if (!string.IsNullOrEmpty(parseResults.ValueForOption<string>("-p")))
```

Comment:
I don't know how the command line parsing works here.  Have you validated that if `--project` is used, the warning isn't printed (since they are both the same `Option`)?  Should we have an automated test for that?

Reply:
You're right. It prints a warning in either case.